### ISSUE TITLE
`selectTemplate` can be called multiple times

### ISSUE THREAD

**Issue by adamdossa** [PolymathNetwork/polymath-core-deprecated#118]
AFAICT this function can be called multiple times (until the point an offering has been set) which would:
  - lock up multiple fees
  - artificially inflate the reputation of Templates

Any reason not to allow this to be set only once?

**Comment by pabloruiz55** [comment]
Duplicate of #56. Thought this was fixed. @satyamakgec ?

**Comment by adamdossa** [comment]
Looks like only the offering contract bit was fixed - I'll fix this in my next PR.
